CRM/Activity/BAO/Activity.php

```diff
@@ -2413,6 +2413,9 @@ public static function checkPermission($activityId, $action) {
       if ($sourceContactId) {
         $allow = CRM_Contact_BAO_Contact_Permission::allow($sourceContactId, $permission);
       }
+      else {
```

Comment:
I think it is a better Idea to keep "elses" less as much as possible so the code  become something like this :

$allow = TRUE;
if ($sourceContactId) {
$allow  = CRM_Contact_BAO_Contact_Permission::allow($sourceContactId, $permission);
}